addons/account/views/account_move_views.xml

```diff
@@ -812,15 +812,20 @@
                                        attrs="{'invisible': [('move_type', 'not in', ('in_invoice', 'in_refund', 'in_receipt'))]}"/>
 
                                 <!-- Invoice payment terms (only invoices) + due date (only invoices / receipts) -->
-                                <label for="invoice_payment_term_id" string="Due Date"
-                                       attrs="{'invisible': [('move_type', 'not in', ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt'))]}"/>
+                                <div class="o_td_label">
+                                    <label for="invoice_payment_term_id" string="Due Date"
+                                           attrs="{'invisible': ['|', ('invoice_payment_term_id', '!=', False), ('move_type', 'not in', ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt'))]}"/>
+                                    <label for="invoice_payment_term_id" string="Payment terms"
+                                           attrs="{'invisible': ['|', ('invoice_payment_term_id', '=', False), ('move_type', 'not in', ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt'))]}"/>
```

Comment:
Reading the commit message:
> - Showing "Payment terms" instead of "Due date" of the user has selected payment terms

I expected `due date` to be replaced by `Payment terms` but it seems you keep payment terms.
Perhaps the commit message isn't completely accurate.

Reply:
Yes we actually just change the label displayed depending on whether the user has chosen a Payment Term or selected a Date.
I.e., by default we show the label "Due Date". However, if the user selects a Payment Term, we hide this label and show the "Payment terms" label instead, so `invoice_payment_term_id` has to be kept

Reply:
Put the invisible domain based on `move_type` on the `<div class="o_td_label"/>`, and the domain based on the `invoice_payment_term_id` on the `<label/>`